### PR TITLE
content-sources: add content-sources config and client

### PR DESCRIPTION
**Pull request by ldjebran**
# Description
To be able to use content-sources repositories we need to create a client and configuration. The first implementation include the ListRepository to allow to retrieve to content sources
 repositories and filter.

Story: THEEDGE-3154

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Comment by ldjebran** [comment]
Ok it's failing because url. JoinPath exist only on version 1.19 and we are using 1.18, fixing ....

File: pkg/clients/repositories/client.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	return url, nil
+}
+
+// ListRepositories return the list of content source repositories
+func (c *Client) ListRepositories(requestParams ListRepositoriesParams, filters ListRepositoriesFilters) ([]Repository, error) {
+	url, err := c.GetBaseURL()
+	if err != nil {
+		return nil, err
+	}
+	repositoriesRawURL := url.String() + "/v1/repositories/"
```
Comment (omaciel) [inline pkg/clients/repositories/client.go:96]: Do we want to hardcode `v1` or should that be a constant?

File: pkg/clients/repositories/client.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+		c.log.WithField("url", baseURL).Error("failed to parse content-sources base url")
+		return nil, err
+	}
+	return url, nil
+}
+
+// ListRepositories return the list of content source repositories
+func (c *Client) ListRepositories(requestParams ListRepositoriesParams, filters ListRepositoriesFilters) ([]Repository, error) {
+	url, err := c.GetBaseURL()
+	if err != nil {
```
Comment (omaciel) [inline pkg/clients/repositories/client.go:93]: Could we add a test for when `c.GetBaseURL()` returns an error?

File: pkg/clients/repositories/client.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+		return nil, err
+	}
+	repositoriesRawURL := url.String() + "/v1/repositories/"
+	repositoriesURL, err := url.Parse(repositoriesRawURL)
+	if err != nil {
```
Comment (omaciel) [inline pkg/clients/repositories/client.go:98]: Another test opportunity where we can also check the data being logged

File: pkg/clients/repositories/client.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+	req, _ := http.NewRequest("GET", requestURL, nil)
+	req.Header.Add("Content-Type", "application/json")
+	headers := clients.GetOutgoingHeaders(c.ctx)
+	for key, value := range headers {
+		req.Header.Add(key, value)
+	}
+
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if err != nil {
```
Comment (omaciel) [inline pkg/clients/repositories/client.go:135]: Test opportunity

File: pkg/clients/repositories/client.go [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+		c.log.WithField("error", err.Error()).Error("content source repositories request error")
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
```
Comment (omaciel) [inline pkg/clients/repositories/client.go:142]: Test opportunity

**Review by omaciel — APPROVED** [review]
/lgtm

There are some test opportunities which might(?) be worthwhile pursuing just to give us a higher level of confidence

**Review by mgold1234 — APPROVED** [review]
/lgtm

File: pkg/clients/repositories/client.go (hunk 1 above)
Comment (ldjebran) [reply pkg/clients/repositories/client.go]: done

File: pkg/clients/repositories/client.go (hunk 3 above)
Comment (ldjebran) [reply pkg/clients/repositories/client.go:113]: done

File: pkg/clients/repositories/client.go (hunk 4 above)
Comment (ldjebran) [reply pkg/clients/repositories/client.go:150]: done

File: pkg/clients/repositories/client.go (hunk 5 above)
Comment (ldjebran) [reply pkg/clients/repositories/client.go:157]: done

**Review by omaciel — APPROVED** [review]
/lgtm

File: pkg/clients/repositories/client.go (hunk 2 above)
Comment (ldjebran) [reply pkg/clients/repositories/client.go:108]: done

**Pull request merged**